modules/system/checks.nix

```diff
@@ -72,16 +72,24 @@ let
         printf >&2 'compatibility with macOS Sequoia 15. Your _nixbld1 user currently has\n'
         printf >&2 'UID %d rather than the new default of 351.\n' "$firstBuildUserID"
         printf >&2 '\n'
-        printf >&2 'You can automatically migrate your users using the following script\n'
-        printf >&2 'from the Nix repository:\n'
+        printf >&2 'You can automatically migrate the users with the following command:\n'
         printf >&2 '\n'
-        printf >&2 '    https://github.com/NixOS/nix/raw/master/scripts/sequoia-nixbld-user-migration.sh\n'
-        printf >&2 '\n'
-        printf >&2 'This should work even if you installed Nix with the Determinate\n'
-        printf >&2 'Systems installer or are using Lix. If you are comfortable using the\n'
-        printf >&2 'script without review, you can run:\n'
-        printf >&2 '\n'
-        printf >&2 "    curl --proto '=https' --tlsv1.2 -sSf -L https://github.com/NixOS/nix/raw/master/scripts/sequoia-nixbld-user-migration.sh | bash -\n"
+        if [[ -e /nix/receipt.json ]]; then
+            if
+                ${pkgs.jq}/bin/jq --exit-status \
+                'try(.planner.settings | has("enable_flakes"))' \
+                /nix/receipt.json \
+                >/dev/null
```

Comment:
Should `stderr` be redirected to `/dev/null` as well?

Reply:
It shouldn’t print anything to standard error unless `/nix/receipt.json` contains malformed JSON. I figured that’s a messed‐up enough situation that you might want some signal of it.